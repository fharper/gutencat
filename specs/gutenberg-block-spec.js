import {
	activatePlugin,
	createNewPost,
	deactivatePlugin,
	enablePageDialogAccept,
	getEditedPostContent,
	insertBlock,
} from '@wordpress/e2e-test-utils';

jest.setTimeout( 60000 );

describe( 'Gutencat', () => {
	const plugin = 'gutencat';

	beforeAll( async () => {
		await enablePageDialogAccept();
	} );

	beforeEach( async () => {
		await activatePlugin( plugin );
		await createNewPost();
	} );

	afterEach( async () => {
		await deactivatePlugin( plugin );
	} );

	it( 'Gutencat should be available', async () => {
		await insertBlock( 'Gutencat' );

		// Check if block was inserted
		expect(
			await page.$( '[data-type="dev-fred-gutencat/gutencat"]' )
		).not.toBeNull();

		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );
} );
