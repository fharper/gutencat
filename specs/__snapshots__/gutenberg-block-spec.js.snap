// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gutencat Gutencat should be available 1`] = `"<!-- wp:dev-fred-gutencat/gutencat /-->"`;

exports[`Wrapper block Gutencat block should be available 1`] = `"<!-- wp:dev-fred-gutencat/gutencat /-->"`;
